sdk/storage/storage-blob/samples/typescript/customPipeline.ts

```diff
@@ -0,0 +1,63 @@
+/* 
+ Setup: Enter your storage account name and shared key in main()
+*/
+
+import {
+  BlobServiceClient,
+  SharedKeyCredential,
+  newPipeline,
+  RawTokenCredential,
+} from "../../src"; // Change to "@azure/storage-blob" in your package
+
+async function main() {
+  // Enter your storage account name and shared key
+  const account = "";
+  const accountKey = "";
+
+  // Use SharedKeyCredential with storage account and account key
+  const sharedKeyCredential = new SharedKeyCredential(account, accountKey);
+
+  // Use TokenCredential with OAuth token
+  const tokenCredential = new RawTokenCredential("token");
+  tokenCredential.token = "renewedToken"; // Renew the token by updating token field of token credential
+
+  // Use AnonymousCredential when url already includes a SAS signature
+  // const anonymousCredential = new AnonymousCredential();
+
+  // Use sharedKeyCredential, tokenCredential or anonymousCredential to create a pipeline
+  const pipeline = newPipeline(sharedKeyCredential);
```

Comment:
Please pass some options to `newPipeline`: like retry options, and telemetry options.

Reply:
Those are already in the `advanced.ts`.

Hmmm.. it makes sense to have those options in this `customPipeline.ts` respecting the file name.